odl/space/npy_tensors.py

```diff
@@ -255,25 +258,51 @@ def __init__(self, shape, dtype=None, **kwargs):
                                      '`exponent`: {} != {}'
                                      ''.format(weighting.exponent, exponent))
                 self.__weighting = weighting
+
+            elif np.isscalar(weighting):
+                if self.ndim == 1:
+                    # Prefer per-axis weighting if possible, it behaves better
```

Comment:
Well yes, I was thinking about this, but I didn't quite dare to remove it completely. What if a user wants constant weighting but the origin of the weights is out of their control? It would be hard to force them to distribute them over the axes in some random way.

But I guess in 99 % of the cases, per axis is what you want, so I'd be okay with removing it. That would also make the code simpler.

Reply:
I guess the "obvious" way to do const weighting would be scaling according to the cell size in each dimension? E.g. if `cell_sides=[1, 2]` and `const=0.5`, I would expect `by_axis_weighting=[0.5, 1.0]`?

Reply:
That would be an *additional* constant factor on the weights. Remember that weightings are defined on the `TensorSpace` which doesn't know anything about cell sides and such.

Reply:
That's right. Not quite sure how to handle this then, I'm not convinced fully removing const as an input option would be a good idea (since it increases the "minimum" syntax needed)

Reply:
Yes, that's the main reason I kept it. Maybe the two can coexist for now. `ConstWeighting` is then the inferior variant that doesn't behave as nicely with indexing. But it's also not used much any more, since `DiscreteLp` now defaults to `PerAxisWeighting` with `cell_sides`, which is what you want in 99 % of the cases.

Reply:
Some new insights into this: `ConstWeighting` has some weird interactions with `PerAxisWeighting`, in particular when doing outer products or reductions.

Here's a new suggestion: Merge `ArrayWeighting` and `ConstWeighting` into `PointwiseWeighting`, since they actually mean the same thing. That name would also be better, more in line with `PerAxisWeighting`.